### PR TITLE
Claim and Do X components

### PR DESCRIPTION
**Pull request by Bean-Sama**

### CODE REVIEW

**Comment by silochad** [comment]
Changes requested from my first review:

-[ ] Check Discord thread for action items and drop a ✅ when completed
-[ ] Constrain the right column to a fixed width, right now it expands if you type large numbers that result in large outputs. In this screenshot, the side-by-side TokenOutputs are causing the issue.
image

File: projects/ui/src/components/Common/SelectionIndicator.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      {selected ? (
+        <Check
+          sx={{
+            color: BeanstalkPalette.white,
+            width: size * SIZE_MULTIPLIER,
+            height: 'auto',
+          }}
+        />
+      ) : null}
+    </Stack>
```
Comment (silochad) [inline projects/ui/src/components/Common/SelectionIndicator.tsx:49]: Looks like it's just the check icon in the top right? This strikes me as a lot of code for just the icon. Could it be reduced down to a CheckCircle with a set color?

https://mui.com/material-ui/material-icons/?query=check&selected=CheckCircle

File: projects/ui/src/components/Common/Form/TokenSelectDialog.tsx [hunk 2]
```diff
@@ -85,6 +114,13 @@ const TokenSelectDialog : TokenSelectDialogC = React.memo(({
     }
   }, [selectedInternal]);
 
+  const filteredTokenList = useMemo(() => {
```
Comment (silochad) [inline projects/ui/src/components/Common/Form/TokenSelectDialog.tsx:117]: Add comment: `// ETH can only be used from EXTERNAL balance`

File: projects/ui/src/components/Common/Form/TokenInputField.tsx [hunk 3]
```diff
@@ -52,6 +57,10 @@ export type TokenInputCustomProps = {
    *
    */
   allowNegative?: boolean;
+  /**
+   * whether to use the wrapped variant or the default text field variant
+   */
+  inputVariant?: 'default' | 'wrapped'
```
Comment (silochad) [inline projects/ui/src/components/Common/Form/TokenInputField.tsx:63]: Do we need to add support for the old variant? Wondering if we can just pull this and have the new `wrapped` version propagate across the site.

**Comment by silochad** [comment]
- [ ] Bug: in this screenshot I have 491 BEAN in my Circulating balance but it appears when I select the Farm mode
image

**Review by silochad — CHANGES_REQUESTED** [review]

File: projects/ui/src/components/Common/SelectionIndicator.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import { remToPx } from '~/util/UI';
+
+export type SelectionIndicatorProps = {
+  selected: boolean;
+  size?: keyof typeof FontSize | number;
+} & BoxProps;
+
+const SIZE_MULTIPLIER = 0.75;
+
+const SelectionIndicator: React.FC<SelectionIndicatorProps> = ({
```
[the remark this answers is not in the document]
Comment (Bean-Sama) [reply projects/ui/src/components/Common/SelectionIndicator.tsx:17]: just removed this component all together

File: projects/ui/src/components/Common/Form/TokenInputField.tsx (hunk 3 above)
Comment (Bean-Sama) [reply projects/ui/src/components/Common/Form/TokenInputField.tsx:63]: Yup! Can do!

**Comment by silochad** [comment]
In some cases, looks like the select dropdown button is off-center:
image

**Pull request closed without merging**